File: scratch/update_settings.js
const dbPath = 'C:\\Users\\bin-g\\AppData\\Roaming\\البصمة الذكية\\pos_data.db';
const db = require('better-sqlite3')(dbPath);

const vatNumber = '311354901300003';
const businessName = 'بوفية نهر الكنوز للوجبات السريعة';

const stmt = db.prepare('UPDATE business_settings SET value = ? WHERE key = ?');

// Use transaction for safe multi-update
db.transaction(() => {
    // If keys don't exist, this might do nothing depending on schema, but usually it exists.
    // Let's do INSERT OR REPLACE or check first.
    
    // Check and update/insert VAT
    const hasVat = db.prepare('SELECT 1 FROM business_settings WHERE key = "vat_number"').get();
    if (hasVat) {
        db.prepare('UPDATE business_settings SET value = ? WHERE key = "vat_number"').run(vatNumber);
    } else {
        db.prepare('INSERT INTO business_settings (key, value) VALUES (?, ?)').run('vat_number', vatNumber);
    }
    
    // Check and update/insert business_name_ar
    const hasName = db.prepare('SELECT 1 FROM business_settings WHERE key = "business_name_ar"').get();
    if (hasName) {
        db.prepare('UPDATE business_settings SET value = ? WHERE key = "business_name_ar"').run(businessName);
    } else {
        db.prepare('INSERT INTO business_settings (key, value) VALUES (?, ?)').run('business_name_ar', businessName);
    }
})();

console.log('Database business settings updated successfully.');
console.log('VAT Number:', vatNumber);
console.log('Business Name:', businessName);

File: scratch/update_settings.py
import sqlite3

db_path = r"C:\Users\bin-g\AppData\Roaming\البصمة الذكية\pos_data.db"
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

vat_number = '311354901300003'
business_name = 'بوفية نهر الكنوز للوجبات السريعة'

# Check and update VAT
cursor.execute("SELECT 1 FROM business_settings WHERE key = 'vat_number'")
if cursor.fetchone():
    cursor.execute("UPDATE business_settings SET value = ? WHERE key = 'vat_number'", (vat_number,))
else:
    cursor.execute("INSERT INTO business_settings (key, value) VALUES (?, ?)", ('vat_number', vat_number))

# Check and update Business Name
cursor.execute("SELECT 1 FROM business_settings WHERE key = 'business_name_ar'")
if cursor.fetchone():
    cursor.execute("UPDATE business_settings SET value = ? WHERE key = 'business_name_ar'", (business_name,))
else:
    cursor.execute("INSERT INTO business_settings (key, value) VALUES (?, ?)", ('business_name_ar', business_name))

conn.commit()
conn.close()

print("Database business settings updated successfully.")
print(f"VAT Number: {vat_number}")
print(f"Business Name: {business_name}")
